dm/syncer/data_validator.go

```diff
@@ -447,119 +610,143 @@ func (v *DataValidator) processRowsEvent(header *replication.EventHeader, ev *re
 		return nil
 	}
 
-	targetTable := v.syncer.route(sourceTable)
-	// todo: syncer will change schema in schemaTracker, will there be data race?
-	// todo: what if table is dropped while validator falls behind?
-	tableInfo, err := v.syncer.schemaTracker.GetTableInfo(sourceTable)
+	table, err := v.genValidateTableInfo(sourceTable)
 	if err != nil {
 		if schema.IsTableNotExists(err) {
 			// not a table need to sync
```

Comment:
For TableNotExists, should we record the unmatched tables and let data validator check if there's a DROP TABLE DDL?

Reply:
if the table is dropped in downstream, we cannot validate this case.

will we skip DROP TABLE event but drop it in schma-tracker?

Reply:
If the table is dropped by operations outside DM, DML on that table will cause error. Generally user will not do that. OK to address this behaviour later.

> will we skip DROP TABLE event but drop it in schma-tracker?

currently schema tracker takes effect before the filter.

Reply:
> If the table is dropped by operations outside DM, DML on that table will cause error.

will add it to `restrictions` in this version.